### PR TITLE
Fix some lgtm alerts

### PR DESCRIPTION
**Pull request by jhelie**
Hi,
I just wanted to quickly fix a few alerts flagged up by lgtm.com. These are mostly simple ones but they were a few if statements that likely didn't work as intended.

A total of 371 alerts have been found by lgtm.com, you can enable PR integration for automated PR reviews and flag these in the future before they find their way into the code base.
https://lgtm.com/projects/g/matplotlib/matplotlib/

see here for PR integration details: https://lgtm.com/docs/lgtm/config/pr-integration

I use matplotlib every day so I hope you'll find these useful!

## PR Checklist

- [ ] Has Pytest style unit tests
- [x ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -717,7 +717,6 @@ def set_size_inches(self, w, h=None, forward=True):
         if not all(np.isfinite(_) for _ in (w, h)):
             raise ValueError('figure size must be finite not '
                              '({}, {})'.format(w, h))
-        dpival = self.dpi
```
Comment (WeatherGod) [inline lib/matplotlib/figure.py]: This used to be useful, but it became invalid when the hidpi support was added.

File: examples/pylab_examples/arrow_demo.py [hunk 2]
```diff
@@ -119,8 +119,6 @@ def do_fontsize(k):
 
     arrow_h_offset = 0.25  # data coordinates, empirically determined
     max_arrow_length = 1 - 2*arrow_h_offset
-
-    max_arrow_width = max_arrow_width
```
Comment (WeatherGod) [inline examples/pylab_examples/arrow_demo.py]: Did some digging. This mistake was there from the very beginning.

File: examples/pylab_examples/arrow_demo.py [hunk 3]
```diff
@@ -165,10 +163,9 @@ def draw_arrow(pair, alpha=alpha, ec=ec, labelcolor=labelcolor):
         else:
             length = max_arrow_length
         # set the transparency of the arrow
-        if display == 'alph':
+        if display == 'alpha':
```
Comment (WeatherGod) [inline examples/pylab_examples/arrow_demo.py]: As was this mistake, too. In fact, this code, I think, became the basis for our FancyArrow class. We should see if this mistake exists in that code, too.

**Comment by WeatherGod** [comment]
I couldn't find a similar mistake with `'alph'` anywhere else in the codebase.

**Comment by jhelie** [comment]
@WeatherGod lgtm tracks alerts over commits and provides introduction point info when available. If this info isn't available it is often because we haven't built/analysed far enough in time for the repo history, thereby suggesting that these alerts have been there for a very long time.

A few of these here did have introduction point info, as you can see here: https://lgtm.com/projects/g/matplotlib/matplotlib/snapshot/ef44115ffaf5c75b9c26386a6091d3ada395a72b/files/lib/matplotlib/backends/backend_ps.py#V141

**Comment by WeatherGod** [comment]
That is really neat, and I am absolutely floored that it detected the
misspelling of 'alpha'.

**Comment by jhelie** [comment]
yes we do think lgtm.com is pretty neat but that one I actually found myself when investigating the unnecessary else :) however you could easily write a query looking for likely misspellings on variable names
see https://lgtm.com/docs/ql/about-ql and https://lgtm.com/query/lang:python/  !

**Comment by tacaswell** [comment]
@jhelie Thanks for the patch and congrats on your first merged PR to Matplotlib :tada: !  

Glad Matplotlib is useful to you and hopefully we will hear from you agian.

**Comment by jhelie** [comment]
@tacaswell thanks - and thank you for your many helpful stackoverflow posts :)

**Pull request merged**